src/ClpIrV1Decoder.cpp

```diff
@@ -0,0 +1,208 @@
+
+
+#include <emscripten/emscripten.h>
+#include <emscripten/val.h>
+#include <emscripten/bind.h>
+
+#include <string>
+#include <iostream>
+
+#include "ffi/ir_stream/decoding_methods.hpp"
+#include "spdlog_with_specializations.hpp"
+#include "TimestampPattern.hpp"
+#include "streaming_compression/zstd/Decompressor.hpp"
+#include "ir/LogEventDeserializer.hpp"
+#include "ffi/encoding_methods.hpp"
+
+#ifdef __cplusplus
+#define EXTERN extern "C"
+#else
+#define EXTERN
+#endif
+
+// FIXME: use macro
+enum class LOG_VERBOSITY
+{
+    NONE = 0,
+    TRACE,
+    DEBUG,
+    INFO,
+    WARN,
+    ERROR,
+    FATAL
+};
+
+const char *VERBOSITIES[] = {
+    nullptr,
+    "TRACE",
+    "DEBUG",
+    "INFO",
+    "WARN",
+    "ERROR",
+    "FATAL",
+};
+
+EM_JS(void, insert_row, (emscripten::EM_VAL results_handle, const char *message, int64_t timestamp, int verbosity), {
+    const results = Emval.toValue(results_handle);
+    results.push([ UTF8ToString(message), timestamp, verbosity ]);
+});
+
+class ClpIrV1Decoder
+{
+private:
+    clp::streaming_compression::zstd::Decompressor m_zstd_decompressor;
+    std::vector<clp::ir::LogEvent<clp::ir::four_byte_encoded_variable_t>> m_log_events;
+    const int m_timezone_offset_ms{0 * 3600 * 1000};
+
+    std::unique_ptr<clp::ir::LogEventDeserializer<clp::ir::four_byte_encoded_variable_t>> m_deserializer;
+    clp::TimestampPattern m_ts_pattern;
+
+public:
+    ClpIrV1Decoder(const uintptr_t dataBuffer, const int length)
+    {
+        std::cout << "ClpIrV1Decoder::ClpIrV1Decoder() got buffer of length=" << length << std::endl;
+        m_zstd_decompressor.open(reinterpret_cast<const char *>(dataBuffer), length);
+    }
+
+    int buildIdx()
+    {
+        decodePreamble();
+
+        m_log_events.reserve(500000);
+        int numDecoded = 0;
+        while (true)
+        {
+            auto const result{m_deserializer->deserialize_log_event()};
+            if (result.has_error())
+            {
+                auto const error{result.error()};
+                if (std::errc::result_out_of_range == error)
+                {
+                    std::cerr << "File contains an incomplete IR stream" << std::endl;
+                }
+                else if (std::errc::no_message_available != error)
+                {
+                    std::cerr << "Failed to decompress: "
+                              << error.category().name() << ":" << error.message()
+                              << std::endl;
+                }
+                break;
+            }
+
+            numDecoded++;
+
+            m_log_events.push_back(result.value());
+        }
+
+        return numDecoded;
+    }
+
+    bool decode(const emscripten::val &results, size_t startIdx, size_t endIdx)
+    {
+        for (size_t log_event_idx = startIdx; log_event_idx < endIdx; log_event_idx++)
+        {
+            auto const &log_event{m_log_events[log_event_idx]};
```

Comment:
```suggestion
            auto const& log_event{m_log_events[log_event_idx]};
```
This should be automatically fixed after applying clang-format